dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -268,16 +267,16 @@ def _map_and_reduce_dimension(self, df, column, records):
         return df
 
     @staticmethod
+    @track_timing(timer_stats_collector)
     def _check_null_value_in_unique_dimension_rows(dim_table):
-        dim_with_null = {}
+        dim_with_null = set()
         cols_to_check = {x for x in dim_table.columns if x != "id"}
 
         for col in cols_to_check:
-            null_count = dim_table.filter(F.col(col).isNull()).count()
-            if null_count > 0:
-                dim_with_null[col] = null_count
+            if not dim_table.select(col).filter(f"{col} is NULL").rdd.isEmpty():
```

Comment:
This is a problematic piece of code. `dim_table` for TEMPO has 27 million rows.  Spark frequently fails with memory errors toward the end of this for loop. It appears to be a garbage collection issue. Many online sources suggest enabling the G1GC garbage collector (Java feature). It did fix the issue for me, at least a few times. I feel uncertain about making this the default behavior in Jade. Running this code in Spark local-mode (instead of a standalone cluster) also seems to run better. I'm reluctant to make changes in this area because I don't expect this type of operation to be common. We will normally be processing large on-disk dataframes, not these large in-memory tables.

Reply:
Good revision of the code. I think when I wrote this func, it was more for internally checking that the mappings get applied correctly and the df gets consolidated correctly afterward. The consolidation piece of the code ( in `_map_and_reduce_dimension`) is not fully fleshed out yet. I think we can skip it once that's done.

Reply:
Now we check for NULLs in both load_data and load_data_lookup.